README.md

```diff
@@ -28,15 +28,34 @@ Create a `PgLock.new` instance and call the `lock` method to ensure exclusive ex
 
 ```ruby
 PgLock.new(name: "all_your_base").lock do
-  if open?
-    API.log(circuit_breaker: true, enabled: true, service: service)
-  else
-    open!(error_count)
+  # stuff
+end
+```
+
+Now no matter how many times this code is executed across any number of machines, one block of code will be allowed to execute at a time.
+
+## Session based locking
+
+The postgres lock is unique across different database sessions, if the same session tries to aquire the same lock it will succeed. So while `PgLock` will guarantee unique execution across machines and processes, it will not block the same process (sharing the same connection session) from running. For example while you would think the middle block would not run in this example:
+
+```ruby
+key = "all_your_base"
+PgLock.new(name: key).lock do
+  puts "First block called"
+  PgLock.new(name: key).lock do
```

Comment:
Is this guaranteed to give you the same underlying session? If not, this may actually deadlock, since the second session (for the inner `PgLock`) may wait for the same advisory lock that the session of the outer `PgLock` just acquired.

That is, in #2, I talk about the problem from the perspective of different threads possibly sharing the same underlying session due to the connection pool, but there's also the reverse problem: the same thread/process can be given a *different* connection by the pool (e.g., if the pool uses a round-robin mechanism to hand out connections, and the pool includes more than one connection).